source/validate_id.cpp

```diff
@@ -1177,11 +1176,143 @@ bool idUsage::isValid<SpvOpCopyMemorySized>(const spv_instruction_t* inst,
   return true;
 }
 
-#if 0
 template <>
-bool idUsage::isValid<SpvOpAccessChain>(const spv_instruction_t *inst,
-                                        const spv_opcode_desc opcodeEntry) {}
-#endif
+bool idUsage::isValid<SpvOpAccessChain>(const spv_instruction_t* inst,
+                                        const spv_opcode_desc) {
+  // The result type must be OpTypePointer. Result Type is at word 1.
+  auto resultTypeIndex = 1;
+  auto resultTypeInstr = module_.FindDef(inst->words[resultTypeIndex]);
```

Comment:
It turns out that here we have access to a `spv_instruction_t*` which does not have a `type_id`.
`spv_parsed_instruction_t*` and `Instruction*` do have the `type_id`.

We could do:
```
Instruction* cur_instr = module_.FindDef(inst->resultTypeId);
```
and then replace `inst->words[resultTypeIndex]` with `cur_instr->type_id()`
But that would be more expensive.